### ISSUE TITLE
Possibility to control the link access/creation property lists from the high level interface

### ISSUE THREAD

**Issue by cderemble** [h5py/h5py#2257]
Currently, link access/creation property lists are global instances created when loading the library via `default_lapl` / `default_lcpl`.
It would be cool if individual groups could have the possibility to override these properties. Ideally, these properties would be passed on to groups returned by `__getitem__`

One possible implementation would be to be to override the properties `_lapl` and `_lcpl` in `Group` : they would default to the global property lists and use the ones optionally provided in the constructor.
The `__init__` function of `Group` and `File` would then need to be modified by adding 2 optional arguments: `lapl` and `lcpl`. The rest of the library should not be affected. By default, the behavior of the library should be unchanged.

**Comment by cderemble** [comment]
possible implementation in #2258

**Comment by aragilar** [comment]
It would seem reasonable to me to have externally referenced files default to whatever mode closest matches the the mode of the main one (with some option to override the mode separately at the high level).

**Comment by takluyver** [comment]
I agree that that makes sense, but I'm also confused - the [HDF5 docs](https://portal.hdfgroup.org/display/HDF5/H5P_SET_ELINK_ACC_FLAGS) suggest that this should already be the default:

> The library will normally use the file access flag used to open the parent file as the file access flag for the target file. 

And h5py doesn't seem to override that:

https://github.com/h5py/h5py/blob/06d9d0d095c1e972f938d874a78598ae42b1df8e/h5py/_hl/base.py#L129-L135

Am I missing somewhere that we're setting that to something different? Or is this a bug?

(I'm more and more convinced that making references to separate files transparent, as external links, external datasets or virtual datasets, is a mistake. :thinking: )

**Comment by takluyver** [comment]
At least for me, following an external link from a read-only file seems to do the right thing (the second file is also read-only):

```
In [3]: f1 = h5py.File('foo.h5', 'r')

In [4]: ds = f1['extlink']

In [5]: ds
Out[5]: <HDF5 dataset "a": shape (10, 10), type "<f4">

In [6]: ds.file
Out[6]: <HDF5 file "foo2.h5" (mode r)>
```

**Comment by cderemble** [comment]
My bad: I got it wrong, let me reformulate.

The problem is when you want to modify a file that contains external links to other files that you know wont be modified. For example, let's copy a dataset from an external file via a link: 
```
f = h5py/File('foo.h5', mode='w')
f['ext_data_set'] = h5py.ExternalLink('foo2.h5', 'data_set')
f.create_dataset('new_data_set', data=f['ext_data_set'])
```
In that example, the file `foo2.h5` will be modified on disk even if its data was only read and not written. This is quite annoying when you want to track the last modification time of your files.
Ideally, you are able to specify the opening mode of the external links when opening a new file, like this:
```
f = h5py/File('foo.h5', mode='w', open_ext_links_in_readonly_mode=True)
f['ext_data_set'] = h5py.ExternalLink('foo2.h5', 'data_set')
f.create_dataset('new_data_set', data=f['ext_data_set'])
```
